Src/coffeescript/cql-execution/src/elm/quantity.coffee

```diff
@@ -19,25 +19,72 @@ module.exports.Quantity = class Quantity extends Expression
     "#{@value} '#{@unit}'"
 
   sameOrBefore: (other) ->
-    if other instanceof Quantity and other.unit == @unit then @value <= parseFloat other.value  else null
+    if other instanceof Quantity and other.unit == @unit
+      @value <= parseFloat other.value
+    else if other instanceof Quantity and other.unit of time_units and @unit of time_units
+      thisSmallestDuration = smallestDuration(@)
+      otherSmallestDuration = smallestDuration(other)
+      thisSmallestDuration <= otherSmallestDuration
+    else null
 
   sameOrAfter: (other) ->
-    if other instanceof Quantity and other.unit == @unit then @value >= parseFloat other.value else null
+    if other instanceof Quantity and other.unit == @unit
+      @value >= parseFloat other.value
+    else if other instanceof Quantity and other.unit of time_units and @unit of time_units
+      thisSmallestDuration = smallestDuration(@)
+      otherSmallestDuration = smallestDuration(other)
+      thisSmallestDuration >= otherSmallestDuration
+    else null
 
   after: (other) ->
-    if other instanceof Quantity and other.unit == @unit then @value > parseFloat other.value else null
+    if other instanceof Quantity and other.unit == @unit
+      @value > parseFloat other.value
+    else if other instanceof Quantity and other.unit of time_units and @unit of time_units
+      thisSmallestDuration = smallestDuration(@)
+      otherSmallestDuration = smallestDuration(other)
+      thisSmallestDuration > otherSmallestDuration
+    else null
 
   before: (other) ->
-    if other instanceof Quantity and other.unit == @unit then @value < parseFloat other.value else null
+    if other instanceof Quantity and other.unit == @unit
+      @value < parseFloat other.value
+    else if other instanceof Quantity and other.unit of time_units and @unit of time_units
+      thisSmallestDuration = smallestDuration(@)
+      otherSmallestDuration = smallestDuration(other)
+      thisSmallestDuration < otherSmallestDuration
+    else null
     
   equals: (other) ->
     if other.unit? && other.value? then @unit == other.unit && @value == parseFloat other.value else null
 
-time_units = {'years':'year',  'months': 'month',  'days' :'day', 'minutes': 'minute', 'seconds':'seconds', 'milliseconds' : 'millisecond' }
+# Hash of time units and their UCUM equivalents
```

Comment:
Why did you change this to return UCUM units?  Previously, it returned the CQL duration units (`year`, `month`, etc).  Since `clean_unit` uses this, and `clean_unit` is used by other functions as well (like `doAddition` and `doSubtraction`), this change may affect other things.

I tried to see if it affected any of the unit tests, but the unit tests don't run at all in this branch.  I think it's the branch you're based on `cql4bonnie` that introduced the unit test problem, however.